hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockServerConnection.java

```diff
@@ -186,8 +188,16 @@ public void close(String msg, Throwable cause) {
                     .getLocalAddressRegistry()
                     .tryRemoveRegistration(remoteUuid, remoteAddress);
         }
+
         if (localConnection != null) {
-            //this is a member-to-member connection
+            assert localNodeEngine != null;
+            Server server = localNodeEngine.getNode().getServer();
+            // this is a member-to-member connection
+            if (server instanceof FirewallingServer) {
+                (((MockServer) ((FirewallingServer) server).delegate)).connectionMap.remove(remoteUuid);
+            } else if (server instanceof MockServer) {
+                ((MockServer) server).connectionMap.remove(remoteUuid);
```

Comment:
I see that in `MockServer.MockServerConnectionManager.MockConnLifecycleListener#onConnectionClose` we do a little more, like closing the remote connection as well and firing connection removed event. Don't we need this here as well?

Reply:
I think when the remote connection is closed with this callback, it will take care of its own cleanup.

Reply:
I meant the other way around. So to fire the connection removed event, closing the other side of the connection here as well.

Reply:
https://github.com/hazelcast/hazelcast/blob/7eef9ec54b2e454efee7042a734521431c3a5bcb/hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockServerConnection.java#L205, here, we send the this event

Reply:
Ah, that line just slipped out from my review window :) Thanks, it's fine than.